sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/SortBasedAggregationIterator.scala

```diff
@@ -131,6 +150,11 @@ class SortBasedAggregationIterator(
         firstRowInNextGroup = currentRow.copy()
       }
     }
+
+    // Serializes the generic object stored in aggregation buffer for TypedImperativeAggregate
+    // aggregation functions.
+    serializeTypedAggregateBuffer(sortBasedAggregationBuffer)
```

Comment:
An alternative approach is to call the serialization just before we output the buffer (https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala#L233-L239). Then, we will not need to check the class at https://github.com/apache/spark/pull/14753/files#diff-9463c978126246071e528ddfa7a376d5R507.